idem_provider_azurerm/exec/azurerm/key_vault/vault.py

```diff
@@ -0,0 +1,568 @@
+        recovered after deletion. Enabling this functionality is irreversible - that is, the property does not accept
+        false as its value. Defaults to False.
+
+    :param enable_purge_protection: A boolean value specifying whether protection against purge is enabled for this
+        vault. Setting this property to true activates protection against purge for this vault and its content - only
+        the Key Vault service may initiate a hard, irrecoverable deletion. The setting is effective only if soft
+        delete is also enabled. Enabling this functionality is irreversible - that is, the property does not accept
+        false as its value.
+
+    :param enabled_for_deployment: A boolean value specifying whether Azure Virtual Machines are permitted to
+        retrieve certificates stored as secrets from the key vault. Defaults to False.
+
+    :param enabled_for_disk_encryption: A boolean value specifying whether Azure Disk Encrpytion is permitted to
+        retrieve secrets from the vault and unwrap keys. Defaults to False.
+
+    :param enabled_for_template_deployment: A boolean value specifying whether Azure Resource Manager is permitted
+        to retrieve secrets from the key vault. Defaults to False.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.key_vault.vault.create_or_update test_name test_rg test_location test_tenant test_sku test_policies \
+              test_uri test_mode test_flags
+
+    '''
+    result = {}
+    vconn = await hub.exec.utils.azurerm.get_client('keyvault', **kwargs)
+
+    sku = {'name': sku}
+
+    # Create the VaultProperties object
+    try:
+        propsmodel = await hub.exec.utils.azurerm.create_object_model(
+            'keyvault',
+            'VaultProperties',
+            tenant_id=tenant_id,
+            sku=sku,
+            access_policies=access_policies,
+            vault_uri=vault_uri,
+            create_mode=create_mode,
+            enable_soft_delete=enable_soft_delete,
+            enable_purge_protection=enable_purge_protection,
+            enabled_for_deployment=enabled_for_deployment,
+            enabled_for_disk_encryption=enabled_for_disk_encryption,
+            enabled_for_template_deployment=enabled_for_template_deployment
+        )
+    except TypeError as exc:
+        result = {'error': 'The object model could not be built. ({0})'.format(str(exc))}
+        return result
+
+    # Create the VaultCreateOrUpdateParameters object
+    try:
+        paramsmodel = await hub.exec.utils.azurerm.create_object_model(
+            'keyvault',
+            'VaultCreateOrUpdateParameters',
+            location=location,
+            properties=propsmodel,
+            **kwargs
+        )
+    except TypeError as exc:
+        result = {'error': 'The object model could not be built. ({0})'.format(str(exc))}
+        return result
+
+    try:
+        vault = vconn.vaults.create_or_update(
+            vault_name = name,
```

Comment:
remove spaces for keyword arguments here (for consistency)

Reply:
my comment was probably too vague, but "remove spaces" was the thing to focus on.
```
vault_name=name,
resource_group_name=resource_group
```
instead of
```
vault_name = name,
resource_group_name = resource_group
```

Reply:
Ah sorry I completely misunderstood that :)